requirements.txt

```diff
@@ -38,8 +37,8 @@ first==2.0.1              # via pip-tools
 funcsigs==1.0.2           # via debtcollector, oslo.utils, pytest
 functools32==3.2.3.post2  # via jsonschema
 futures==3.2.0            # via s3transfer
```

Comment:
I just wonder if we can get rid of functools32 and futures here?

Reply:
will check tomorrow.

Reply:
no, we cannot. they are dependencies introduced by boto3 and python-openstackclient respectively.

Reply:
too bad, because those two keep me out of running tests in py3

Reply:
what do you mean by "running tests"? are these tests teuthology's own tests?

Reply:
I mean:
```
virtualenv -p python3.7
pip install -r requirements.txt 
```
fails for me

Reply:
we need to tell pip that functools and futures are python2 only requrirements.

Reply:
The problem is `pip-compile` itself. When you call it from a python3 env, you get:
```
(venv3) tom@minze:~/devel/ceph/teuthology$ pip-compile -P boto3 2>&1 | functools
```

and from a py2 venv:
```
(venv2) tom@minze:~/devel/ceph/teuthology$ pip-compile -P boto3 2>&1 | grep functools
functools32==3.2.3.post2  # via jsonschema
```

So imo `pip-compile` is broken by design (or there is some flag to fix this...)

Reply:
@kshtsk i feel sorry about your pain. but i think the long term solution is to ditch the python2 support.